File: java/Stack/easy/remove-all-adjacent-duplicates-in-string.java
class Solution {
    public String removeDuplicates(String S) {
       StringBuilder a=new StringBuilder();
        a.append(S);
        for(int i=0;i<S.length() && i!=S.length()-1;i++){
            if (a.charAt(i)==a.charAt(i+1))
            {a.deleteCharAt(i);
            a.deleteCharAt(i+1);                 
            }
        }
        if(S==a.toString()){return S;}
        return removeDuplicates(a.toString());
    }
}

File: cpp/Stack/easy/remove-all-adjacent-duplicates-in-string.cpp
class Solution {
public:
    string removeDuplicates(string S) {
        stack<char> s;
        for(auto i : S) {
            if(s.empty() || s.top() != i) {
                s.push(i);
            } else {
                s.pop();
            }
        }
        string ans = "";
        while(!s.empty()) {
            ans += (s.top());
            s.pop();
        }
        reverse(ans.begin(), ans.end());
        return ans;
    }
};
